services/mailer.test.js

```diff
@@ -0,0 +1,45 @@
+jest.mock('../lib/log')
+const logGen = require('../lib/log')
+const logger = {
+    info: jest.fn(),
+    error: jest.fn()
+}
+logGen.mockReturnValue(logger)
+
+jest.mock('mailgun-js')
+const mailgun = require('mailgun-js');
+const messages = {
+    send: jest.fn().mockImplementation((data) => {
+        if (data.to) return Promise.resolve('hello')
+        return Promise.reject('rejected')
+    })
+}
+mailgun.mockImplementation(() => {
+    return {
+        messages: () => {
+            return messages
+        }
+    }
+})
+const email = require('./mailer');
+
+describe('Test mailgun', ()=>{
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    it('should test if mocksend and mailgun is called', async () => {
+        await email.sendConfirmationEmail('paul@github.com', 'token123')
+        expect(messages.send).toHaveBeenCalledTimes(1)
+        expect(messages.send.mock.calls[0][0]).toMatchSnapshot()
+        expect(logger.info).toHaveBeenCalledTimes(1)
+        expect(logger.info.mock.calls[0][0]).toEqual('Confirmation Email successfully sent')
+    })
+
+    it('should call logger.error when function is called with invalid argument', async () => {
+        await email.sendConfirmationEmail(null, null)
```

Comment:
you should mock reject, I think its important to test here when service is down. like:
```
messages.send = jest.fn().mockReturnValue( Promise.reject('invalid') )
```

Reply:
I think it does the same as what it does in 13-14 lines. I see how it's easier to be read. Is that what you meant?